File: Java/Bitwise/program13.java
//Accept number from user and check whether First nibble is ON or OFF

import java.util.*;

class Bitwise
{
    public boolean CheckNibble(int iNo)
    {
        int iMask = 0X0000000f;
        int iResult = 0;

        iResult = iNo & iMask;

        if(iResult == 0)
        {
            return false;
        }
        else
        {
            return true;
        }
    }
}

class program13
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        int iValue = sobj.nextInt();

        Bitwise bobj = new Bitwise();

        boolean bRet = bobj.CheckNibble(iValue);
        if(bRet == true)
        {
            System.out.println("First nibble is completely on");
        }
        else
        {
            System.out.println("First nibble is off");
        }
    }
}

File: C/Bitwise/program13.c
//Accept number from user and check whether First nibble is ON or OFF

#include<stdio.h>
#include<stdbool.h>

bool CheckNibble(int iNo)
{
    int iMask = 0X0000000f;
    int iResult = 0;

    iResult = iNo & iMask;

    if(iResult == 0)
    {
        return false;
    }
    else
    {
        return true;
    }
}

int main()
{
    int iValue = 0;
    bool bRet;

    printf("Enter the number\n");
    scanf("%d",&iValue);

    bRet = CheckNibble(iValue);
    if(bRet == true)
    {
        printf("First Nibble is ON");
    }
    else
    {
        printf("First Nibble is OFF");
    }

    return 0;
}